pkg/controllers/runtime_controller.go

```diff
@@ -91,7 +91,10 @@ func (r *RuntimeReconciler) ReconcileInternal(ctx cruntime.ReconcileRequestConte
 			ctx.Log.Info("the dataset can't be bound to the runtime, because it's already bound to another runtime ",
 				"dataset", dataset.Name)
 			dataset = nil
+			err = fmt.Errorf("the dataset can't be bound to the runtime, because it's already bound to another runtime %s", dataset.Name)
```

Comment:
`dataset.Name` will lead to NPE here for the reason that `dataset` just be set to `nil`

Reply:
Thanks for spotting it!